pkg/ottl/expression.go

```diff
@@ -89,6 +93,25 @@ func (l *listGetter[K]) Get(ctx context.Context, tCtx K) (interface{}, error) {
 	return evaluated, nil
 }
 
+type StandardStringGetter[K any] struct {
```

Comment:
Looking at the code for `Get`, it looks like this should be the same for each of the primitive types. One way we could share code between them is by making a generic `StandardPrimitiveTypeGetter[K any, T any]` that replaces `string` with `T` in the `Get` implementation here. Embedding the `StandardPrimitiveTypeGetter` struct inside the `StandardStringGetter` struct would then allow `StandardStringGetter` to satisfy the  `StringGetter` interface without having to implement its own method, and this process could be extended to the other types. What do you think?

Reply:
oo I like that idea.  I'll give it a try.

Reply:
Looks great. I saw that you also simplified it so we don't even need the `StandardStringGetter` type, nice work there.